Kernel/Memory/PhysicalRegion.cpp

```diff
@@ -123,29 +124,25 @@ RefPtr<PhysicalPage> PhysicalRegion::take_free_page()
 
 void PhysicalRegion::return_page(PhysicalAddress paddr)
 {
-    size_t full_size_zone_index = (paddr.get() - lower().get()) / large_zone_size;
-    size_t large_zone_count = m_pages / (large_zone_size / PAGE_SIZE);
-
-    if (full_size_zone_index < large_zone_count) {
-        auto& zone = m_zones[full_size_zone_index];
-        VERIFY(zone.contains(paddr));
-        zone.deallocate_block(paddr, 0);
-        if (m_full_zones.contains(zone))
-            m_usable_zones.append(zone);
-        return;
-    }
+    auto large_zone_base = lower().get();
+    auto small_zone_base = lower().get() + (m_large_zones * large_zone_size);
 
-    for (size_t i = large_zone_count; i < m_zones.size(); ++i) {
-        auto& zone = m_zones[i];
-        if (zone.contains(paddr)) {
-            zone.deallocate_block(paddr, 0);
-            if (m_full_zones.contains(zone))
-                m_usable_zones.append(zone);
-            return;
-        }
+    auto base = large_zone_base;
+    size_t divisor = large_zone_size;
+    size_t index_base = 0;
+
+    if (paddr.get() >= small_zone_base) {
+        base = small_zone_base;
+        divisor = small_zone_size;
+        index_base = m_large_zones;
     }
 
-    VERIFY_NOT_REACHED();
+    size_t zone_index = index_base + (paddr.get() - base) / divisor;
```

Comment:
one thing that is unfortunate here is that this division is very expensive, 
my approach and the previous one had division by a constant expression, which is up to 100x faster in that case, due to being replaces with a single `&` by the compiler.
It would be nice if you could get optimization back.

```suggestion
size_t zone_index;
if (paddr.get() < small_zone_base)
    zone_index = (paddr.get() - large_zone_base) / large_zone_size;
else 
    zone_index = m_large_zones + (paddr.get() - small_zone_base) / small_zone_size;
```

This also reduces the code size quite a bit